src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2882,7 +2883,14 @@ void LinearScan::SetContainsAVXFlags(unsigned sizeOfSIMDVector /* = 0*/)
         compiler->GetEmitter()->SetContainsAVX(true);
         if (sizeOfSIMDVector == 32)
         {
-            compiler->GetEmitter()->SetContains256bitAVX(true);
+            compiler->GetEmitter()->SetContains256bitOrMoreAVX(true);
+        }
+    }
+    if (compiler->canUseEvexEncoding())
+    {
+        if (compiler->compExactlyDependsOn(InstructionSet_AVX512F) && (sizeOfSIMDVector == 64))
+        {
+            compiler->GetEmitter()->SetContains256bitOrMoreAVX(true);
```

Comment:
Should there be some `SetContainsAVX512(true)` to match the `SetContainsAVX(true)` concept?

Reply:
I do not think so. This is mostly used for inserting `vzeroupper `in my recollection. And I'm not sure we need to differentiate between 256 and 512 for that. Please feel free to correct me if I'm wrong

Reply:
That looks to be right, at a glance. Sounds good to me.